### ISSUE TITLE
Provide more context in Pages explorer menu items for screen reader users

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#5335]
### Issue summary

In the Pages explorer menu, users can view a page, edit it, view its child pages. Those buttons are correctly labeled for screen readers to announce, but the labels do not provide enough context – they should announce the name of the page in the button label, for example "View child pages of Gallery", or "Edit Breads".

<img width="694" alt="pages-explorer-menu" src="https://user-images.githubusercontent.com/877585/58544084-86813d00-81f8-11e9-944d-6d7629e98812.png">

### Steps to Reproduce

1. Go to the Wagtail admin and open the explorer menu
2. Have the screen reader announce the links within the menu

**Comment by thibaudcolas** [comment]
Addressed in #5386.
